independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/AnalyticsService.java

```diff
@@ -0,0 +1,260 @@
+        if (this.userId.isNew()) { // fixme when inactive on 1st call it will not send identity.
+            this.restClient.postIdentity(Identity.builder()
+                    .userId(this.userId.getUuid())
+                    .context(context)
+                    .timestamp(Instant.now())
+                    .build());
+        }
+    }
+
+    Map<String, Object> createContextMap(ApplicationModel applicationModel,
+            Map<String, Object> buildInfo) {
+        WorkspaceModuleId moduleId = applicationModel.getApplicationModule().getId();
+
+        return new ContextBuilder()
+                .mapPair(PROP_APP)
+                .pair(PROP_NAME, hashSHA256(moduleId.getGroupId() + ":" + moduleId.getArtifactId()))
+                .pair(PROP_VERSION, hashSHA256(moduleId.getArtifactId() + ":" + moduleId.getVersion()))
+                .build()
+                .mapPair(PROP_JAVA)
+                .pair(PROP_VENDOR, getProperty("java.vendor", "N/A"))
+                .pair(PROP_VERSION, getProperty("java.version", "N/A"))
+                .build()
+                .mapPair(PROP_GRAALVM)
+                .pair(PROP_VENDOR, ofNullable(buildInfo.get(GRAALVM_VERSION_DISTRIBUTION)).orElse("N/A"))
+                .pair(PROP_VERSION, ofNullable(buildInfo.get(GRAALVM_VERSION_VERSION)).orElse("N/A"))
+                .pair(PROP_JAVA_VERSION, ofNullable(buildInfo.get(GRAALVM_VERSION_JAVA)).orElse("N/A"))
+                .build()
+                .mapPair(PROP_BUILD)
+                .pair(PROP_MAVEN_VERSION, ofNullable(buildInfo.get(MAVEN_VERSION)).orElse("N/A"))
+                .pair(PROP_GRADLE_VERSION, ofNullable(buildInfo.get(GRADLE_VERSION)).orElse("N/A"))
+                .build()
+                .mapPair(PROP_QUARKUS)
+                .pair(PROP_VERSION, getQuarkusVersion(applicationModel))
+                .build()
+                .pair(PROP_IP, VALUE_NULL_IP)
+                .mapPair(PROP_LOCATION)
+                .pair(PROP_COUNTRY, Locale.getDefault().getCountry())
+                .build()
+                .mapPair(PROP_OS)
+                .pair(PROP_NAME, getProperty("os.name", "N/A"))
+                .pair(PROP_VERSION, getProperty("os.version", "N/A"))
+                .pair(PROP_OS_ARCH, getProperty("os.arch", "N/A"))
+                .build()
+                .mapPair(PROP_CI)
+                .pair(PROP_CI_NAME, getBuildSystemName())
+                .build()
+                .mapPair(PROP_KUBERNETES)
+                .pair(PROP_DETECTED, isKubernetesDetected())
+                .build()
+                .pair(PROP_TIMEZONE, ZoneId.systemDefault().getDisplayName(TextStyle.NARROW, Locale.ENGLISH))
+                .build();
+    }
+
+    private String isKubernetesDetected() {
+        return Boolean.toString(allEnvSet(
+                "KUBERNETES_SERVICE_HOST",
+                "KUBERNETES_SERVICE_PORT"));
+    }
+
+    private String getBuildSystemName() {
+        String travis = getenv("TRAVIS");
```

Comment:
Is it worth to check for the `CI` env as well and add it to the result as a "boolean"?

Reply:
it wasn't a requirement. Sorry.